apps/chat_websocket/lib/chat_websocket/channels/room_channel.ex

```diff
@@ -130,6 +132,41 @@ defmodule ChatWebsocket.RoomChannel do
     end
   end
 
+  def handle_in("room:archive", _params, socket) do
+    %{room: room, user: user} = socket.assigns
+
+    with {:ok, record} <- UpdateRoom.call(room.id, user, "archived"),
+         :ok <- broadcast!(socket, "room:archived", record),
+         {:ok, _} <- maybe_update_support_presence(record),
+         subscriptions <- ListSubscriptions.call(record),
+         :ok <- broadcast_user_subscriptions!(subscriptions) do
+      reply(:ok, %{room: record}, socket)
+    else
+      _ -> reply(:error, "Error archiving room", socket)
+    end
+  end
+
+  defp maybe_update_support_presence(room) do
+    case room.type do
+      "support" -> TrackSupportRooms.update(room, %{state: room.state})
+      _ -> {:ok, true}
+    end
+  end
+
+  def handle_in("room:activate", _params, socket) do
```

Comment:
Let's keep the action name consistent across the client and server code. Instead of `room:activate` let's call it `room:reactivate`. Definitely see why `room:activate` was chosen in the first place, but consistency the client and server codebases is more important than naming consistency with room states on the server side.

Note: this will require updating the client too: https://github.com/oberon-chat/chat-client/pull/24/files#diff-5f01ec77da1add7cca26942825450464R116